flake-module.nix

```diff
@@ -81,9 +88,27 @@ in
   config = {
     perSystem = { config, self', inputs', pkgs, ... }:
       let
+        # Like pkgs.runCommand but runs inside nix-shell with a mutable project directory.
+        #
+        # It currenty respects only the nativeBuildInputs (and no shellHook for
+        # instance), which seems sufficient for our purposes. We also set $HOME and
+        # make the project root mutable, because those are expected when running
+        # something in a project shell (it is indeed the case with HLS).
+        runCommandInNixShell = devShell: projectRoot: name: attrs: command:
+          pkgs.runCommand name (attrs // { buildInputs = devShell.nativeBuildInputs; })
+            ''
+              # Copy project root to a mutable area
+              # We expect "command" to mutate it.
+              export HOME=$TMP
+              cp -R ${projectRoot} $HOME/project
+              chmod -R a+w $HOME/project
+              pushd $HOME/project
+
+              ${command} > $out
```

Comment:
This is leaky. For example, if `command` is `foo; bar`, only `bar` is redirected.

I'd also make the caller responsible for creating the `$out`.